lib/flux-core/src/iter/traits/iterator.rs

```diff
@@ -16,7 +16,10 @@ defs! {
 trait Iterator {
     #[spec(
         fn(self: &mut Self[@curr_s]) -> Option<Self::Item>[!<Self as Iterator>::done(curr_s)]
-        ensures self: Self{next_s: <Self as Iterator>::step(curr_s, next_s)}
+        ensures self: Self{next_s:
```

Comment:
@vrindisbacher why do you need this change? 

Technically, the `next` is a _relation_ and for each `impl` you can define it so that IF the `done(curr_s)`  is true then `next(curr_s, next_s)` only holds if `next_s == curr_s`; so we shouldn't really need to change the spec at all (or if we do, it should suffice to just modify the `next` for the appropriate `impl`?